docs/core/integration-quality-scale/rules/icon-translations.md

```diff
@@ -53,6 +57,49 @@ class MySensor(SensorEntity):
 }
 ```
 
+### Range-based icons
+
+For numeric entities, you can define icons that change based on numeric ranges.
+This is particularly useful for sensors like battery levels or signal strength indicators.
+
+In this example, we define a battery sensor that changes its icon based on the battery percentage.
+The ranges must be numeric values in ascending order.
+
+`icons.json`
+
+```json
+{
+  "entity": {
+    "sensor": {
+      "battery_level": {
+        "default": "mdi:battery",
+        "range": {
+          "0": "mdi:battery-outline",
+          "10": "mdi:battery-10",
+          "20": "mdi:battery-20",
+          "30": "mdi:battery-30",
+          "40": "mdi:battery-40",
+          "50": "mdi:battery-50",
+          "60": "mdi:battery-60",
+          "70": "mdi:battery-70",
+          "80": "mdi:battery-80",
+          "90": "mdi:battery-90",
+          "100": "mdi:battery"
+        }
+      }
+    }
+  }
+}
+```
+
+When using range-based icons:
+
+- The range values must be numeric and in ascending order
+- The icon for a given value will be chosen from the highest range value that's less than or equal to the entity's current value
+- Both integer ("0", "100") and decimal ("0.5", "99.9") range values are supported
+- The default icon is used for values outside the defined ranges, for example, when the entity is unavailable or has an invalid value
+- If there are also state-based icons defined, those will take precedence over the range-based icons
```

Comment:
In my opinion we should add also here the example from the blog post

Reply:
Done! Extended the wording 👍